### PR TITLE
Cleanup and increase coverage of DisplayFormatAttribute

### PR DESCRIPTION
**Pull request by hughbe**
Before #if netcoreapp was always false, so these tests never ran

### CODE REVIEW

File: src/System.ComponentModel.Annotations/tests/DisplayFormatAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        [Fact]
+        public void ConvertEmptyStringToNull_Set_GetReturnsExpected()
+        {
+            var attribute = new DisplayFormatAttribute { ConvertEmptyStringToNull = false };
+            Assert.False(attribute.ConvertEmptyStringToNull);
+        }
+
+        [Fact]
+        public void HApplyFormatInEditMode_Set_GetReturnsExpected()
```
Comment (stephentoub) [inline src/System.ComponentModel.Annotations/tests/DisplayFormatAttributeTests.cs]: Nit: 'H'?

Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/DisplayFormatAttributeTests.cs]: Accidental typo, fixed

**Comment by lajones** [comment]
@hughbe. I liked the previous set up - it seems to me the better solution would be to set netcoreapp (or some other variable) to true under the appropriate circumstances. Is it some kind of build convention that makes the tests in the `DisplayFormatAttributeTests.netcoreapp.cs` file only run under the appropriate circumstances?

**Comment by hughbe** [comment]
It's my understanding that the convention in corefx is to split it by file

**Comment by lajones** [comment]
@hughbe Is there something that ensures that tests ending with `.netcoreapp.cs` are only run in a `netcoreapp` setting?

**Comment by lajones** [comment]
Never mind previous comment - just spotted the `Condition` in the .csproj.

**Comment by hughbe** [comment]
Oh sorry I misread, yeah there's a condition in the csproj

**Comment by hughbe** [comment]
The linux failures are unrelated, but let's get a clean CI
@dotnet-bot Test Portable Linux x64 Debug Build
@dotnet-bot Test Portable Linux x64 Release Build

**Comment by hughbe** [comment]
@dotnet-bot Portable Linux x64 Release Build

**Comment by hughbe** [comment]
@dotnet-bot Test Portable Linux x64 Release Build

**Comment by danmosemsft** [comment]
@lajones good to merge?

**Comment by lajones** [comment]
@ajcvickers This looks good to me. But did you want to look at this before I mark it approved?

**Comment by ajcvickers** [comment]
@lajones @danmosemsft Looks good to me too. I approved it.

**Comment by danmosemsft** [comment]
@ajcvickers thanks

**Pull request merged**